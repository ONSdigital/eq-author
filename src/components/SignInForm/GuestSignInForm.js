import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import Button from "components/Button";
import { partial } from "lodash";

const displayName =
  process.env.NODE_ENV === "development" ? "Guesty McGuestFace" : "Guest";

const GUEST_USER = {
  displayName,
  email: "guest@example.org"
};

const GuestSignInButton = styled(Button)`
  margin-top: 1em;
`;

const GuestSignInForm = ({ onSignIn }) => (
  <GuestSignInButton primary onClick={partial(onSignIn, GUEST_USER)}>
    Sign in as Guest
  </GuestSignInButton>
);

GuestSignInForm.propTypes = {
  onSignIn: PropTypes.func.isRequired
};

export default GuestSignInForm;
