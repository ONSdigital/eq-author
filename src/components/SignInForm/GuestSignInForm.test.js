import React from "react";
import GuestSignInForm from "./GuestSignInForm";
import { shallow } from "enzyme";

describe("GuestSignInForm", () => {
  it("should render", () => {
    expect(shallow(<GuestSignInForm onSignIn={jest.fn()} />)).toMatchSnapshot();
  });

  it("should invoke callback on sign in", () => {
    const handleSignIn = jest.fn();
    const wrapper = shallow(<GuestSignInForm onSignIn={handleSignIn} />);

    wrapper.simulate("click");

    expect(handleSignIn).toHaveBeenCalled();
  });
});
