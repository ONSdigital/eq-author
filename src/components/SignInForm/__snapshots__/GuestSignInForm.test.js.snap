// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GuestSignInForm should render 1`] = `
<GuestSignInForm__GuestSignInButton
  onClick={[Function]}
  primary={true}
>
  Sign in as Guest
</GuestSignInForm__GuestSignInButton>
`;
